pkg/operator2/workload/workload_controller.go

```diff
@@ -0,0 +1,308 @@
+package workload
+
+import (
+	"context"
+	"fmt"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	openshiftconfigclientv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	clusteroperatorv1helpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	appsv1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+	"time"
+)
+
+const (
+	workQueueKey              = "key"
+	workloadDegradedCondition = "WorkloadDegraded"
+)
+
+type syncOperatorFunc func() (*appsv1.DaemonSet, []error)
+
+type OAuthAPIServerOperator struct {
```

Comment:
`s/OAuthAPIServerOperator/OAuthAPIServerWorkloadController`? Or maybe just `APIServerWorkloadController`

Reply:
yes, I should change the name.